sql/sql_plugin.cc

```diff
@@ -4028,45 +4047,69 @@ int unlock_plugin_data()
   Create deep copy of system_variables instance.
 */
 struct system_variables *
-copy_system_variables(const struct system_variables *src,
-                      bool enable_plugins)
+copy_system_variables(THD *thd, bool enable_plugins)
 {
   struct system_variables *dst;
+  ulong idx;
 
-  DBUG_ASSERT(src);
+  DBUG_ASSERT(thd);
 
   dst= (struct system_variables *)
     my_malloc(sizeof(struct system_variables), MYF(MY_WME | MY_FAE));
-  *dst = *src;
+  *dst = thd->variables;
 
   if (dst->dynamic_variables_ptr)
```

Comment:
I see no issues here (at least in 5.6). THD-local variables are initialized lazily when they are accessed. I don't see any way to access them from another thread.